CredentialProvider.Microsoft/Program.cs

```diff
@@ -208,23 +208,38 @@ public static async Task<int> Main(string[] args)
             }
         }
 
-        internal static async Task RunNuGetPluginsAsync(IPlugin plugin, ILogger logger, TimeSpan timeout, CancellationToken cancellationToken)
+        internal static async Task WaitForPluginExitAsync(IPlugin plugin, ILogger logger, TimeSpan shutdownTimeout)
```

Comment:
Would it be useful to have the opportunity to set the timeout as an environment variable and have 2 minutes as a default?